### DISCUSSION TITLE
How do you install onedrive?

### DISCUSSION THREAD

**Discussion by daleamon** [abraunegg/onedrive#2561] [Q&A] [answered]
I've got the tar.gz unpacked and it is not at all clear how to build it. README.md does not give the sequence of build commands. The fact that the Makefile and others have a .in implies they are to be run through a pre-processor.  One would expect that install.sh would do it, but it doesn't seem to know what to do. Even ./install.sh --help throws a wobbly. .install.sh appears to want to be given a SRCDIR but there is no explanation of what it considers the SRCDIR to be. Is it '."? Is it src? The next step would be to go inside install.sh with emacs and try to figure out what it wants, but it is near end of day on Friday...

**Comment by abraunegg** [comment, marked as answer]
@daleamon 

> I've got the tar.gz unpacked and it is not at all clear how to build it. README.md does not give the sequence of build commands. The fact that the Makefile and others have a .in implies they are to be run through a pre-processor. One would expect that install.sh would do it, but it doesn't seem to know what to do. Even ./install.sh --help throws a wobbly. .install.sh appears to want to be given a SRCDIR but there is no explanation of what it considers the SRCDIR to be. Is it '."? Is it src? The next step would be to go inside install.sh with emacs and try to figure out what it wants, but it is near end of day on Friday...

What are you talking about as this makes zero sense.

There is no install.sh .... 

As for the instructions to build from source - I guess you missed this section in the readme.md ?

https://github.com/abraunegg/onedrive#documentation-and-configuration-assistance

**Comment by daleamon** [comment]
```
onedrive-2.4.25$ ls
aclocal.m4    config     configure.ac  docs        LICENSE      onedrive.1.in  src    version
CHANGELOG.md  configure  contrib       install-sh  Makefile.in  README.md      tests
```

**Reply by daleamon** [reply]
okay, it -sh not .sh. I still don't know how to make it do the install.

**Comment by abraunegg** [comment]
@daleamon 
Please read the documentation - you have 2 options:

1. Install via your distribution package
2. Compile and install manually following the specific instructions for your distribution

**Comment by daleamon** [comment]
Yes, I did go through it, and I missed part of it due to the messy formatting in emacs. I'd done the one section and missed the other further along. I'm surprised they don't have a markdown display mode. Or maybe they do and I haven't got it.

Compile and install seem to have proceeded.

**Comment by abraunegg** [comment]
@daleamon 

> Yes, I did go through it, and I missed part of it due to the messy formatting in emacs. 

There is this: 

https://jblevins.org/projects/markdown-mode/

**Comment by daleamon** [comment]
Thanks. Will look into that.
